### ISSUE TITLE
valid dataset

### ISSUE THREAD

**Issue by junkangwu** [jin530/LOCA#2]
Hi, nice work!
    But I found it is lacking validation dataset while test dataset as the measurement of early stop? Is it normal? Looking forword to your relpy~

**Comment by jin530** [comment]
Hello,
Thanks for the question.

We checked several other implementations below for experimentation. 
And like other implementations, we used the test set to check the accuracy during training VAE and LOCA_VAE.

EASE and LOCA_EASE are solved with a closed-form solution, so we used the test set once after the training for them.

deeprec: https://github.com/cheungdaven/DeepRec
neurec: https://github.com/wubinzzu/NeuRec
ncf: https://github.com/hexiangnan/neural_collaborative_filtering
ngcf: https://github.com/xiangwang1223/neural_graph_collaborative_filtering

**Comment by junkangwu** [comment]
Thanks for your reply. But when I refer to the similar work which leverages vae for recommandation such as Multi-VAE, RecVae and Macrivae. The obey the same rule for dataset split and utilize the valid dataset as the measurement of early stop. In my opinion, test dataset should not viewed during the training time. In that way, they are not allowed to be the sign of early stop?

Multi-VAE: https://github.com/dawenl/vae_cf
RecVAE: https://github.com/ilya-shenbin/RecVAE
Macrivae: https://jianxinma.github.io/disentangle-recsys.html

**Comment by junkangwu** [comment]
And 
- May I ask you about the best hyper parameters settings in each dataset? such as local model number etc.

**Comment by jin530** [comment]
Sorry for the late reply.
As you said, I agree that looking at the test set only once is more appropriate to check the model's generality, and it would be good to experiment with the LOCA_VAE one more time.
(As mentioned above, in the case of LOCA_EASE, the result is not changed.)

The hyperparameters are below:

number of local models: 300

[LOCA_VAE]
ML10M: train_h 1.0 test_h 0.8
ML20M: train_h 0.6 test_h 0.3
AMusic: train_h 1.5 test_h 0.7
AGames: train_h 1.0 test_h 0.6
Yelp: train_h 1.0 test_h 0.5

[LOCA_EASE]
ML10M: train_h 1.0 test_h 0.9
ML20M: train_h 0.7 test_h 0.5
AMusic: train_h 1.2 test_h 0.5
AGames: train_h 1.2 test_h 0.4
Yelp: train_h 1.2 test_h 0.5
